apps/cnquery/cmd/builder/builder.go

```diff
@@ -355,7 +355,7 @@ func dockerProviderCmd(commonCmdFlags commonFlagsFn, preRun commonPreRunFn, runF
 		Use:    "docker ID",
 		Short:  docs.GetShort("docker"),
 		Long:   docs.GetLong("docker"),
-		Args:   cobra.ExactArgs(1),
+		Args:   cobra.ExactArgs(0),
```

Comment:
This will break existing behaviour:

```
cnquery shell docker centos
```

You would need to set this to `cobra.MaximumNArgs(1)`. Then you would need to add additional logic. If the no specific argument was provided the discover flag needs to be set.

Reply:
I think we need to somehow unify the commands. So if I want to open a shell to a container I need to do:
```
cnquery shell docker centos
```
but if I want to scan the container I need to do:
```
cnquery shell docker container centos
```
Shouldn't these be identical?

Reply:
I agree that there are some inconsistencies that we should iron out.

- it is confusing that container and image scan are on different levels
- ` cnquery shell container b62b276baab6` for running container
- `cnquery shell container image ubuntu:20.04` for image

For docker it is more clear:

```
    cnquery shell docker b62b276baab6 => container id or image id 
    cnquery shell docker container b62b => only container id
    cnquery shell docker image ubuntu:latest => only image id
```